pennylane/compiler.py

```diff
@@ -0,0 +1,227 @@
+
+        if name in cls.__BACKENDS:
+            cls.__BACKENDS.remove(name)
+
+        return False
+
+    @classmethod
+    def available_backends(cls) -> List[str]:
+        """Return the available compiler packages."""
+        return cls.__BACKENDS
+
+    @classmethod
+    def active(cls) -> bool:
+        """Check whether the caller is inside a QJIT evaluation context.
+
+        Return:
+            bool : True if the caller is inside a QJIT evaluation context
+        """
+        if not cls.__BACKENDS:
+            raise RuntimeError("There is no available compiler package.")
+
+        if "cpl_utils" not in cls.__ENTRY_POINTS:
+            raise RuntimeError("There is no available 'cpl_utils' entry point.")
+
+        utils_loader = cls.__ENTRY_POINTS["cpl_utils"].load()
+        return utils_loader.contexts.EvaluationContext.is_tracing()
+
+    @classmethod
+    def qjit(cls, fn=None, *args, **kwargs):
+        """A just-in-time decorator for PennyLane and JAX programs using Catalyst.
+
+        This decorator enables both just-in-time and ahead-of-time compilation,
+        depending on whether function argument type hints are provided.
+
+        .. note::
+
+            Currently, ``lightning.qubit`` is the only supported backend device
+            for Catalyst compilation. For a list of supported operations, observables,
+            and measurements, please see the :doc:`/dev/quick_start`.
+
+        Args:
+            fn (Callable): the quantum or classical function
+            autograph (bool): Experimental support for automatically converting Python control
+                flow statements to Catalyst-compatible control flow. Currently supports Python ``if``,
+                ``elif``, ``else``, and ``for`` statements. Note that this feature requires an
+                available TensorFlow installation.
+            target (str): the compilation target
+            keep_intermediate (bool): Whether or not to store the intermediate files throughout the
+                compilation. If ``True``, intermediate representations are available via the
+                :attr:`~.QJIT.mlir`, :attr:`~.QJIT.jaxpr`, and :attr:`~.QJIT.qir`, representing
+                different stages in the optimization process.
+            verbosity (bool): If ``True``, the tools and flags used by Catalyst behind the scenes are
+                printed out.
+            logfile (Optional[TextIOWrapper]): File object to write verbose messages to (default -
+                ``sys.stderr``).
+            pipelines (Optional(List[Tuple[str,List[str]]])): A list of pipelines to be executed. The
+                elements of this list are named sequences of MLIR passes to be executed. A ``None``
+                value (the default) results in the execution of the default pipeline. This option is
+                considered to be used by advanced users for low-level debugging purposes.
+
+        Returns:
+            catalyst.QJIT object.
```

Comment:
Out of scope right now, but if we do support additional backends, we will need to update this docstring to be more agnostic :)